dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/merge/MergeProcessor.java

```diff
@@ -0,0 +1,773 @@
+package io.openliberty.microprofile.openapi20.merge;
+
+import static java.util.stream.Collectors.toList;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.BiPredicate;
+
+import org.eclipse.microprofile.openapi.OASFactory;
+import org.eclipse.microprofile.openapi.models.Components;
+import org.eclipse.microprofile.openapi.models.ExternalDocumentation;
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.Operation;
+import org.eclipse.microprofile.openapi.models.PathItem;
+import org.eclipse.microprofile.openapi.models.Paths;
+import org.eclipse.microprofile.openapi.models.info.Info;
+import org.eclipse.microprofile.openapi.models.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.models.servers.Server;
+import org.eclipse.microprofile.openapi.models.tags.Tag;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.websphere.ras.annotation.Trivial;
+
+import io.openliberty.microprofile.openapi20.MergedOpenAPIProvider;
+import io.openliberty.microprofile.openapi20.OpenAPIProvider;
+import io.openliberty.microprofile.openapi20.utils.MessageConstants;
+import io.openliberty.microprofile.openapi20.utils.OpenAPIModelVisitor;
+import io.openliberty.microprofile.openapi20.utils.OpenAPIModelWalker;
+import io.smallrye.openapi.api.util.MergeUtil;
+import io.smallrye.openapi.runtime.io.Format;
+import io.smallrye.openapi.runtime.io.OpenApiSerializer;
+
+/**
+ * Merges multiple OpenAPI documents together
+ * <p>
+ * Merging OpenAPI documents is not trivial. In particular the following things have to be considered:
+ * <ul>
+ * <li>To ensure paths don't overlap, prepend the context root to the front of paths where possible and update any server urls appropriately
+ * <li>If paths still overlap, documents cannot be merged
+ * <li>Unless it's the same across all documents, move servers and security requirements from the top level to under each path
+ * <li>Rename tags, component names and operationIds where necessary to ensure they don't overlap
+ * </ul>
+ *
+ */
+public class MergeProcessor {
+
+    private static final TraceComponent tc = Tr.register(MergeProcessor.class);
+
+    private static final Info MERGED_INFO;
+
+    static {
+        MERGED_INFO = OASFactory.createInfo();
+        MERGED_INFO.setTitle("Merged documentation");
```

Comment:
NLS?

Reply:
I don't think so, we don't NLS the other default values for the OpenAPI model. I will make it a constant.